Sources/_NIOConcurrency/AsyncSequenceSupport.swift

```diff
@@ -0,0 +1,136 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=5.5)
+import struct NIO.ByteBuffer
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public struct NIOByteBufferToUInt8AsyncSequence<Upstream: AsyncSequence>: AsyncSequence where Upstream.Element == ByteBuffer {
```

Comment:
I don't love the name. `URLSession` uses `AsyncBytes` (i.e. `URLSession.AsyncBytes`) which I think is quite clear.  How about `ByteBuffer.AsyncBytes`?

Reply:
Is this not basically an async version of FlattenSequence/.joined() from the standard library, specialised for ByteBuffer?

So... ByteBuffer.FlattenSequence, perhaps?